pkg/controller/k8/cluster.go

```diff
@@ -149,6 +150,7 @@ func (k *Cluster) GetServicesWithLabel(ctx context.Context, namespace string, la
 	}
 	labelSelector := labels.SelectorFromSet(labelMap)
 	options := &client.ListOptions{
+		Namespace: namespace,
```

Comment:
The changes here in this file add the namespace filter. Previously, the argument of the function `namespace` was not used at all in the function body, resulting in wrong results in the caller.

Reply:
Good catch. We need this fix.